python/source/schema.rst

```diff
@@ -108,4 +108,87 @@ as far as they are compatible
     pyarrow.Table
     col1: int32
     col2: string
-    col3: double
\ No newline at end of file
+    col3: double
+
+Merging multiple schemas
+========================
+
+When you have multiple separate groups of data that you want to combine
+it might be necessary to unify their schemas to create a superset of them
+that applies to all data sources.
+
+.. testcode::
+
+    import pyarrow as pa
+
+    first_schema = pa.schema([
+        ("country", pa.string()),
+        ("population", pa.int32())
+    ])
+
+    second_schema = pa.schema([
+        ("country_code", pa.string()),
+        ("language", pa.string())
+    ])
+
+:func:`unify_schemas` can be used to combine multiple schemas into
+a single one:
+
+.. testcode::
+
+    union_schema = pa.unify_schemas([first_schema, second_schema])
+
+    print(union_schema)
+
+.. testoutput::
+
+    country: string
+    population: int32
+    country_code: string
+    language: string
+
+If the combined schemas have overlapping columns, they can still be combined
+as far as the colliding columns retain the same type (``country_code``):
```

Comment:
Small suggestion (but take it or leave it :)): I would personally directly show an example where there is a column in common, instead of splitting it in two examples first without common column and then with. 
Having no columns in common at all doesn't seem like the typical use case for this, so directly showing the second example can simplify the cookbook entry a bit (one example less to show), while you can still explicitly mention how the common column is handled.

Reply:
I actually went in the direction of showing the case where there are no overlapping columns first because I thought that when talking about "merging" two things it's easier to get what's going on if you see a perfect union of the two original collections.

Reply:
Yes, but as far as I understood the unify function, its primary intended use case is to unify slightly different schemas (eg different ordering, missing column, ..), and not to merge schemas in a typical "join" context where you join two different tables (although this is of course also possible). 
So while the second use might be easier to explain in itself, IMO it doesn't really help overall to explain the typical use of `unify_schemas`. 
(but as I said, take it or leave it :), those things will always be inherently subjective (so I should maybe not have reacted again :))